src/sys/mman.rs

```diff
@@ -41,6 +41,9 @@ libc_bitflags! {
         MAP_FILE;
         /// Share this mapping. Mutually exclusive with `MAP_PRIVATE`.
         MAP_SHARED;
+        /// Force mmap to check and fail on unknown flags. This also enables `MAP_SYNC`.
+        #[cfg(all(any(target_arch = "x86"), not(linux_android), not(target_os = "hurd"), not(target_env = "uclibc"), not(freebsdlike)))]
```

Comment:
```suggestion
        /// Force mmap to check and fail on unknown flags. This also enables `MAP_SYNC`.
        #[cfg(target_os = "linux")]
```

This flag is available on all the Linux targets.